elooplib/client.cpp

```diff
@@ -1,131 +1,187 @@
 
-Client::Client(int descr) : Node(descr){}
-Client::Client(Client &&rhs) : Node(static_cast<Node &&>(rhs)),
-                               request_field(std::move(rhs.request_field)) {}
+Client::Client(int descr) : Node(descr) {}
+Client::Client(Client &&rhs)
+    : Node(static_cast<Node &&>(rhs)),
+      request_field(std::move(rhs.request_field)),
+      response(std::move(rhs.response)) {}
 
 int Client::handleConnection()
 {
-    char buff[512];
-    size_t rcv;
-
-    rcv = recv(getFd(), buff, 512, 0);
-    if (rcv < 0)
-    {
-        return -1;
-    }
-    if (rcv == 0)
-    {
-        Node::closeConnection();
-        return 0;
-    }
-    if (rcv > 0)
+  char buff[512];
+  size_t rcv;
+
+  rcv = recv(getFd(), buff, 512, 0);
+  if (rcv < 0)
+  {
+    return -1;
+  }
+  if (rcv == 0)
+  {
+    Node::closeConnection();
+    return 0;
+  }
+  int res = 0;
+  if (rcv > 0)
+  {
+    request_field.append(buff, rcv);
+    auto p = request_field.find('\n');
+    if (p != std::string::npos)
     {
-        request_field.append(buff, rcv); 
-        auto p = request_field.find('\n');
-        if (p != std::string::npos)
-        {
-
-            response = request_field.substr(0, p + 1);
-            sendData();
-            request_field.clear();
-        }
+
+      response = request_field.substr(0, p + 1);
+      res = sendData();
+      request_field.clear();
     }
-    return 0;
+  }
+  return res;
 }
 
-void Client::recognizeData()
+int Client::recognizeData()
 {
-    std::string filter{" \n"};
-
-    auto pos = response.find_first_of(filter);
-    std::string command = response.substr(0, pos + 1);
-
-    if (command == "time\n")
-    {
-        Client::handleTime();
-        return;
-    }
-
-    if (command == "echo ")
-    {
-        Client::handleEcho();
-        return;
-    }
+  if (response.starts_with("time\n"))
+  {
+    return Client::handleTime();
+  }
+
+  if (response.starts_with("echo "))
+  {
+    return Client::handleEcho();
+  }
+
+  if (response.starts_with("compress "))
+  {
+    return Client::handleCompress();
+  }
+  else
+  {
+    response = "Wrong request\n";
+    return 0;
+  }
+}
 
-    if (command == "compress ")
-    {
-        Client::handleCompress();
-        return;
-    }
-    else
-    {
-        response = "Wrong request\n";
-    }
+int Client::handleEcho()
+{
+  response = response.substr(5, request_field.size() - 5);
+  return 0;
 }
 
-void Client::handleEcho()
+int Client::handleTime()
 {
-    response = response.substr(5, request_field.size() - 5);
+  auto t = std::chrono::system_clock::now();
+  auto tt = std::chrono::system_clock::to_time_t(t);
+  response = std::ctime(&tt);
+  return 0;
 }
 
-void Client::handleTime()
+int Client::handleCompress()
+{
+  response = response.substr(9, request_field.size() - 10);
+  const std::filesystem::path dir(response);
+  if (!exists(dir))
+  {
+    response = "Path is not exists\n";
+    return 0;
+  }
+  int fd = eventfd(0, EFD_CLOEXEC);
+  return fd;
```

Comment:
Why do you return fd? return value of all functions is treated as if thread pool needs to be used. Return 1 instead. Or better use bool as a return value.

Reply:
i need fd, to create a new Node obj(ClientTask) in eventloop

Reply:
The why don't you create it just near ` Node obj(ClientTask)` in eventloop? You don't use fd here at all

Reply:
i need a fd to add it to pollfd. My point is `handleConnection` returns zero, or fd for new `client`( whatever it is)
I have no other idea, how to create a new client, without return code from `handleConnection`

Reply:
My argument, as well as many others was about style, not the correctness. I'll send you a voice memo with explanations